app/src/main/java/com/kickstarter/ui/activities/NewsletterActivity.kt

```diff
@@ -1,13 +1,83 @@
+import com.kickstarter.libs.utils.SwitchCompatUtils
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.models.User
+import com.kickstarter.ui.data.Newsletter
+import com.kickstarter.viewmodels.NewsletterViewModel
+import rx.android.schedulers.AndroidSchedulers
 
-class NewsletterActivity : AppCompatActivity() {
+@RequiresActivityViewModel(NewsletterViewModel.ViewModel::class)
+class NewsletterActivity : BaseActivity<NewsletterViewModel.ViewModel>() {
+
+    private lateinit var binding: ActivityNewsletterBinding
+
+    private lateinit var currentUserType: CurrentUserType
+    private lateinit var build: Build
+    private lateinit var ksString: KSString
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_newsletter)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_newsletter)
+
+        this.build = environment().build()
+        this.currentUserType = environment().currentUser()
+        this.ksString = environment().ksString()
+
+        this.viewModel.outputs.user()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::displayPreferences)
+
+        this.viewModel.errors.unableToSavePreferenceError()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { ViewUtils.showToast(this, getString(R.string.profile_settings_error)) }
+
+        this.viewModel.outputs.showOptInPrompt()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::showOptInPrompt)
+
+        binding.apply {
+            happeningSwitch.setOnClickListener { viewModel.inputs.sendHappeningNewsletter(happeningSwitch.isChecked) }
+            newsEventsSwitch.setOnClickListener { viewModel.inputs.sendPromoNewsletter(newsEventsSwitch.isChecked) }
+            projectsWeLoveSwitch.setOnClickListener { viewModel.inputs.sendWeeklyNewsletter(projectsWeLoveSwitch.isChecked) }
+        }
+    }
+
+    private fun displayPreferences(@NonNull user: User) {
+        SwitchCompatUtils.setCheckedWithoutAnimation(this.binding.happeningSwitch, isTrue(user.happeningNewsletter()))
+        SwitchCompatUtils.setCheckedWithoutAnimation(this.binding.newsEventsSwitch, isTrue(user.promoNewsletter()))
+        SwitchCompatUtils.setCheckedWithoutAnimation(this.binding.projectsWeLoveSwitch, isTrue(user.weeklyNewsletter()))
+    }
+
+    private fun newsletterString(newsletter: Newsletter): String? {
+        return when (newsletter) {
+            Newsletter.HAPPENING -> getString(R.string.profile_settings_newsletter_happening)
+            Newsletter.PROMO -> getString(R.string.profile_settings_newsletter_promo)
+            Newsletter.WEEKLY -> getString(R.string.profile_settings_newsletter_weekly)
```

Comment:
I think we're missing the Games newsletter.

Reply:
Yes I added it in this second PR I have coming to add the rest of the Newsletter switches. At the time when I started this PR Danny didn't have Games in the wireframes, so I have it in the next PR wired up to go.